### PR TITLE
Implement binary support

### PR DESCRIPTION
**Pull request by dominikschulz**
This PR adds a binary command with several subcommands that streamline
working with binary data. To retain compatability with other password
managers the data is store base64 encoded.

We may want to add confirmation interaction for some of the destructive operations in the future, but right now I wanted to keep this PR to the point. Those can easily and cleanly added later on.

### CODE REVIEW

File: action/binary.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	default:
+		return fmt.Errorf("ambiquity detected. Unhandled case. Please report a bug")
+	}
+}
+
+func (s *Action) binaryValidate(buf []byte, name string) error {
+	h := sha256.New()
+	_, _ = h.Write(buf)
+	fileSum := fmt.Sprintf("%x", h.Sum(nil))
+
```
Comment (zwopir) [inline action/binary.go]: I suggest extracting the hash summing, so we can change or extend it easily on one place

File: action/binary.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	var err error
+	buf, err = s.binaryGet(name)
+	if err != nil {
+		return err
+	}
+	_, _ = h.Write(buf)
+	storeSum := fmt.Sprintf("%x", h.Sum(nil))
+
+	if fileSum != storeSum {
+		return fmt.Errorf("Hashsum mismatch (file: %s, store: %s)", fileSum, storeSum)
```
Comment (zwopir) [inline action/binary.go]: if we plan to expose this function (via wrapper) to the UI, we should return a custom error type, so we can handle error in retrieving a store secret and a validation mismatch differently

File: action/binary.go (hunk 1 above)
Comment (dominikschulz) [inline action/binary.go]: Hmm, I considered it, but in this case I'm just comparing two hashsums in one method. The hashum doesn't leave this method, so I'm inclined to leave it as is. We can easily change this later without affecting our API.

File: action/binary.go (hunk 2 above)
Comment (dominikschulz) [inline action/binary.go]: Indeed, but we'll have to do a big round of error handling / propagation refactoring anyway so I'd like to postpone this. I don't want to introduce one kind of error propagation here that may differ from the way we decide it later.

But definitely a good point, we should keep this in mind.

**Pull request merged**